### PR TITLE
New dialect AWS Athena

### LINKED ISSUE #2168: New dialect: AWS Athena
Is there interest in adding a dialect for [AWS Athena](https://docs.aws.amazon.com/athena/latest/ug/ddl-sql-reference.html)?

As a workaround one can get somewhere by applying hive dialect on Athena queries, but it's not perfect.

So, probably a lot of code could be borrowed from hive dialect implementation.

For DML queries Athena dialect is basically a subset of Presto. But SQLFluff doesn't currently support Presto dialect either.

### PR DESCRIPTION
**Pull request by cmotta**
### What does it do?
Adds AWS Athena dialect.

### Any helpful background information?
[AWS Athena Docs](https://docs.aws.amazon.com/athena/latest/ug/what-is.html)
DDL is based on [Hive](https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DDL).
DML is based on [PrestoDB](https://prestodb.io/docs/0.217/index.html) 

### Any new dependencies? Why were they added?
No

### Does it close any issues?
Fixes #2168

### Are there any other side effects of this change that we should be aware of?
No

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  ✅`.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).

- Added appropriate documentation for the change. 
This PR does not change any docs. 

- Created GitHub issues for any relevant followup/future enhancements if appropriate

### CODE REVIEW

**Comment by barrywhart** [comment]
From a brief review, this looks good to me. Let's wait until @tunetheweb can review it before merging.

File: src/sqlfluff/dialects/dialect_athena.py
Comment (barrywhart) [inline src/sqlfluff/dialects/dialect_athena.py:36]: Can this be removed?

File: src/sqlfluff/dialects/dialect_athena.py
Comment (barrywhart) [inline src/sqlfluff/dialects/dialect_athena.py:31]: IIUC, Presto has a lot of overlap with Hive. Should we consider basing Presto dialect on Hive rather than ANSI? (Not a recommendation, just a question.)

File: src/sqlfluff/dialects/dialect_athena.py
Comment (barrywhart) [inline src/sqlfluff/dialects/dialect_athena.py:221]: What do comments like this mean? Does this mean `INT` is not a valid Presto data type? If so, do we plan to remove things like this?

File: src/sqlfluff/dialects/dialect_athena.py
Comment (barrywhart) [inline src/sqlfluff/dialects/dialect_athena.py:243]: Does this work? Usually things like this are modeled as a sequence of keywords, not a single string.

File: src/sqlfluff/dialects/dialect_athena.py
Comment (barrywhart) [inline src/sqlfluff/dialects/dialect_athena.py:504]:
```suggestion
    """An `UNLOAD` statement.
```

File: src/sqlfluff/dialects/dialect_athena.py
Comment (cmotta) [reply src/sqlfluff/dialects/dialect_athena.py]: Thanks, left it by mistake.

File: src/sqlfluff/dialects/dialect_athena.py
Comment (cmotta) [reply src/sqlfluff/dialects/dialect_athena.py]: Thanks for point it out. 

Just added a sequence for these types and created a specific fixture testing it.

File: src/sqlfluff/dialects/dialect_athena.py
Comment (cmotta) [reply src/sqlfluff/dialects/dialect_athena.py]: Thx

File: src/sqlfluff/dialects/dialect_athena.py
Comment (cmotta) [reply src/sqlfluff/dialects/dialect_athena.py]: Thanks for pointing this out. Both engines support `int`. 
Also removed inline comments.

File: src/sqlfluff/dialects/dialect_athena.py
Comment (cmotta) [reply src/sqlfluff/dialects/dialect_athena.py:31]: I thought about that, but I think I was biased by tunetheweb last comment on issue #2168 and chose to start with ansi and build from there. Maybe going forward I can experiment using hive as base.

File: src/sqlfluff/dialects/dialect_athena.py
Comment (tunetheweb) [inline src/sqlfluff/dialects/dialect_athena.py:36]: You’re updating these without clearing them.

That’s OK, and will likely result in less errors as the ANSI piece of grammar you are not overriding will still work and will not require overriding to set them to `Nothing()`

However, the downside of this is all ANSI reserved words will still be reserved words in anthena. Only a few of these so probably not a problem.
But also all the unreserved words will be treated as unreserved and there’s a bigger list of those that will only be allowed to be used as identifiers if quoted.

It’s probably fine to leave as is (especially as an initial commit) and many of the dialects are done like this for simplicity reasons), but just wanted to make you aware of the downsides.

**Review by tunetheweb — APPROVED** [review]
Looks pretty good to me. A few minor comments and things to consider.

**Review by tunetheweb — CHANGES_REQUESTED** [review]
Actually I think there’s a few more minutes places we need to add the dialect:
- The main project README
- The docs/source/dialects
- The setup.py files (for SEO on pypi - though don’t think that works to be honest!)

File: src/sqlfluff/dialects/dialect_athena_keywords.py
Comment (cmotta) [reply src/sqlfluff/dialects/dialect_athena_keywords.py]: Thanks for pointing this out. I have changed all keywords to upper case to maintain consistency.

File: src/sqlfluff/dialects/dialect_athena_keywords.py
Comment (cmotta) [reply src/sqlfluff/dialects/dialect_athena_keywords.py]: Done!

File: test/fixtures/dialects/athena/create_database.sql
Comment (cmotta) [reply test/fixtures/dialects/athena/create_database.sql]: Done!

File: src/sqlfluff/dialects/dialect_athena_keywords.py
Comment (cmotta) [reply src/sqlfluff/dialects/dialect_athena_keywords.py]: Done!

File: src/sqlfluff/dialects/dialect_athena.py
Comment (cmotta) [reply src/sqlfluff/dialects/dialect_athena.py:36]: Thanks a lot for clarifying! 
Going forward I will probably take some time to evaluate if we should clear at least the unreserved keywords.

File: src/sqlfluff/dialects/dialect_athena.py
Comment (cmotta) [reply src/sqlfluff/dialects/dialect_athena.py:31]: Nice. Good to know that this can be done.

**Review by tunetheweb — APPROVED** [review]

**Comment by tunetheweb** [comment]
Great first PR! Thanks for that. We’ll include on 1.2.0 in next few weeks.

**Comment by cmotta** [comment]
Thank you all for maintaining this project. Looking forward to the release.

**Pull request merged**
